peps/pep-0747.rst

```diff
@@ -442,14 +440,29 @@ so must be disambiguated based on its argument type:
 -  As a value expression, ``Annotated[x, ...]`` has type ``object``
    if ``x`` has a type that is not ``type[C]`` or ``TypeExpr[T]``.
 
-**Union**: The type expression ``T1 | T2`` is ambiguous with the value ``int1 | int2``,
-so must be disambiguated based on its argument type:
+**Union**: The type expression ``T1 | T2`` is ambiguous with 
+the value ``int1 | int2``, ``set1 | set2``, ``dict1 | dict2``, and more,
+so must be disambiguated based on its argument types:
+
+-  As a value expression, ``x | y`` has type equal to the return type of ``type(x).__or__``
+   if ``type(x)`` overrides the ``__or__`` method.
+
+   -  When ``x`` has type ``builtins.type``, ``types.GenericAlias``, or the
```

Comment:
I'm trying to say `x | y` is evaluated (still) as a normal value expression. If the type of `x` is `type` or `GenericAlias` then the signature in typeshed will be used, which will say that a `TypeExpr[X | Y]` is returned, as described in §["Changed signatures"](https://github.com/python/peps/pull/3856#changed-signatures).

> What if the __or__ method is present, but evaluating it generates a type error (e.g. because y's type is incompatible with the signature)?

Use whatever behavior is used now.

Reply:
> I'm trying to say x | y is evaluated (still) as a normal value expression.

Evaluated by whom? A static type checker? If so, the expression will _never_ be evaluated as `GenericAlias` because that's not something a static type checker knows or cares about (nor should it).

Reply:
Based on your later comments, I now understand that at least pyright special-cases the `|` operator rather than using typeshed's definitions (for `__or__` and `__ror__`) and using the regular rules for calling an overloaded method.

If there's a desire to *continue* special-casing the `|` operator, I might need some help from you to **transcribe all the current rules for the `|` operator** (which I doubt are in any specification) so that I can effectively propose a minimal diff to those rules.